idris.nix

```diff
@@ -0,0 +1,61 @@
+{ mkDerivation, aeson, annotated-wl-pprint, ansi-terminal
```

Comment:
Is this output from cabal2nix, or is it hand-written? If it's generated code, should it be in the repo, or should the repo contain instructions to generate it? I'm not sure what best practices are around that.

Reply:
It is the output from `cabal2nix` with some hand-edits to make sure `perl`, `nodejs` and `cabal` are available and to fix some problems with `fsnotify` (IIRC) that forgets to add some frameworks to the propagated dependencies. I'm a newbie with Nix, so I don't have an answer for the best practices.

Reply:
@shlevy , I seem to recall you being an experienced Nixer. Is this the right way to do things? (sorry if I remembered wrong)

Reply:
I don't think we've quite reached a point yet where there are best practices here exactly, but my preferred approach would be:

a) Autogenerate the cabal-based Nix expression on-the-fly with `callCabal2nix`
b) Add any necessary modifications using `overrideCabal`, `overrideDerivation`, etc. after importing the base generated expression.

This won't be helpful to you now, but we're developing a set of libraries internally at Target that we're hoping to open source over the next few months that will make this workflow clean and easy and hopefully get it adopted more widely.

Reply:
I'm happy to make these changes if/when the PR is in an otherwise-acceptable state.

Reply:
Thanks @shlevy! @jacereda, does the approach of importing the unmodified cabal2nix output make sense to you?

Reply:
Sure, I just need to figure out how.